yt/yt/server/scheduler/scheduler.cpp

```diff
@@ -795,11 +796,15 @@ class TScheduler::TImpl
                 operation->GetState()));
         }
 
+        auto error = TError("Suspend operation by user request");
+        if (reason) {
+            error <<= TErrorAttribute("reason", *reason);
+        }
         DoSuspendOperation(
             operation,
-            TError("Suspend operation by user request"),
+            error,
             abortRunningAllocations,
-            /*setAlert*/ false);
+            /*setAlert*/ !!reason);
```

Comment:
`!reason.has_value()` would be more readable

And the question: we intentionally want to set alert if reason was specified? Could you describe some usecase?

Reply:
I was told that the alert will be visible to the user in UI, so it looks like the place to put it.

Reply:
Ok, it is for visibility reasons